### ISSUE TITLE
Kodi crashes immediately when starting to play AV1 on Ubuntu 24.04 noble

### ISSUE THREAD

**Issue by baranyaib90** [xbmc/xbmc#25458]
### Describe the bug
Here is a clear and concise description of what the problem is:

I have downloaded the following youtube video in AV1 format (1080p): https://www.youtube.com/watch?v=m1jY2VLCRmY
It can be played with VLC in other device, but when I try with kodi it crashed with coredump.
I have attached the crashlog it contains the backtrace. If you need some variable or more detailed frame debugging, please let me know, I will keep the coredump (but I can reproduce it constantly). Because of privacy reasons I would not share the coredump.

Interesting thing is, that preview image and video durations are OK. (See attached photo from TV.)

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Kodi should not crash, best would be to play the file.

## Actual Behavior

It crashed with coredump.

## Possible Fix

I'm not sure if this is a ffmpeg or kodi bug, but one is obviously faulty.

### To Reproduce

1. On Ubuntu 24.04 install kodi with apt-get
2. Obtain some AV1 video file, like what I recommended
3. Play the AV1 video file
4. Kodi will likely crash

### Debuglog

The debuglog can be found here:
[kodi_crashlog-20240708_230819.log](https://github.com/user-attachments/files/16134598/kodi_crashlog-20240708_230819.log)

### Screenshots 
Here are some links or screenshots to help explain the problem:

![photo_2024-07-08_23-18-35](https://github.com/xbmc/xbmc/assets/5031516/abc58e17-b3a9-4200-a73a-ed785fab9378)

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

Nothing else.

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Ubuntu 24.04 LTS
 - Kodi version: 2:20.5+dfsg-1build2 (ubuntu package)
 - My GPU of Intel N100 should support hardware decoding of AV1.

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by neo1973** [comment]
This  is most likely fixed with #23996 but that's only for Kodi 21 because we don't support 20 anymore.

**Comment by baranyaib90** [comment]
OK, I understand. I assume I will have to wait ~1 year for ubuntu to use Kodi version 21.
I'm not familiar with flathub, I would not destabilize my system by installing Kodi 21 from there. So I will skip testing it for now.
I'm also trying to ask for backport the fix via patch: https://bugs.launchpad.net/ubuntu/+source/kodi/+bug/2072534
I'm expecting no luck with that...

**Comment by baranyaib90** [comment]
@neo1973 just FYI: the mentioned possible fix above did not change anything.
I got a test build of kodi containing that patch here: https://bugs.launchpad.net/ubuntu/+source/kodi/+bug/2072534 
And it did not work out. Crash has the very same backtrace again.

**Comment by baranyaib90** [comment]
I have upgraded my system to Ubuntu 24.10 which has Kodi 2:21.1+dfsg-4 package available.
Kodi crashes with same backtrace when playing AV1.
Now I have disabled vaapi hardwer avcceleration in case of AV1 format in the settings.
It burns CPU, but does not crashes.

I have also tested with live pendrive and kodi crashed . So it is not my install's fault. Something with HW acceleration.

**Comment by TheFeshy** [comment]
I had this same problem - any AV1 video crashed Kodi when VAAPI decode was enabled for it, with (nearly) the same CPU (n95) on Fedora.

In my case it was fixed by ditching the Fedora "free" versions of FFMEPG and MESA (which do not contain proprietary codecs, though AV1 is free) and explicitly installing the intel drivers (in Fedora, it's intel-media-driver package)

I am not familiar with Unbutu packaging, but your link in the post above this seems to list similar drivers for Ubuntu.

With those changes, I can now play AV1 up to 4k without issue, and 8k just barely makes it over 30fps.

**Comment by baranyaib90** [comment]
@TheFeshy thank you so much for sharing your story! The idea you gave worked for me as well:
I had previously used "intel-media-va-driver" package but now I have replaced with "intel-media-va-driver-non-free".
I turned back vaapi hw acceleration for AV1 as well and it did not crash!
I even have made a few measurements:

```
4k 30fps: Kodi CPU: 18% GPU Render: 10%, GPU Video: 10%
4k 60fps: Kodi CPU: 25% GPU Render: 10%, GPU Video: 30%
8k 30fps: Kodi CPU: 20% GPU Render: 26%, GPU Video: 40-50%
```
Measured with "sudo intel_gpu_top" from intel-gpu-tools package.
CPU/GPU temperatre raised only with 2-3 Celsius, which is nothing.
Without acceleration only 4k 30fps was watchable with 50-60% CPU load. Other two was lagging pretty much.

Used videos:
https://www.youtube.com/watch?v=LXb3EKWsInQ (4k 60fps)
https://www.youtube.com/watch?v=m1jY2VLCRmY (4k 30fps and 8k 30fps)
